### PR TITLE
Display Case Search Errors on Case Management Page

### PR DESCRIPTION
**Pull request by zandre-eng**
## Product Description

If the response is not 200 when fetching case data, the related error message will be shown as an alert to the user:
Screenshot from 2024-07-22 16-08-17

## Technical Summary

Link to ticket [here](https://dimagi.atlassian.net/browse/SC-3792).

If there are errors when Case Search Filters are applied, these will now be shown as an alert to the user. This also extends to any other errors that may occur when fetching case data after case filters have been applied. If there is no response text given in the response, then a generic message will be displayed to the user.

## Feature Flag

`GEOSPATIAL`

### Safety story

- Local testing

This change will only affect domains that have the Geospatial feature flag enabled.

### Automated test coverage

All changes are done in JS, so no automated tests exist.

### QA Plan

No QA planned.

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/geospatial/static/geospatial/js/geospatial_map.js [hunk 1]
```diff
@@ -419,7 +419,15 @@ hqDefine("geospatial/js/geospatial_map", [
         // Hide the datatable rows but not the pagination bar
         $('.dataTables_scroll').hide();
 
-        if (xhr.responseJSON.aaData.length && mapModel.mapInstance) {
+        if (xhr.status !== 200) {
+            if (xhr.responseText.length) {
+                alertUser.alert_user(xhr.responseText, 'danger');
+            } else {
+                alertUser.alertUser(
```
Comment (kaapstorm) [inline corehq/apps/geospatial/static/geospatial/js/geospatial_map.js:426]: This is camel case here, and snake case on line 424. Is that correct?

Comment (ajeety4) [reply corehq/apps/geospatial/static/geospatial/js/geospatial_map.js:426]: Good catch. Looks like this should be snake case

Comment (zandre-eng) [reply corehq/apps/geospatial/static/geospatial/js/geospatial_map.js]: Ah yes, this should indeed be snake case. Addressed in [fe127d4](https://github.com/dimagi/commcare-hq/pull/34895/commits/fe127d4ed96f3d7485f6f02bee533e45e61cc71a).

**Review by ajeety4 — APPROVED** [review]
:+1:

**Review by kaapstorm — APPROVED** [review]

**Pull request merged**